File: java/believe/gui/testing/FakeGuiContext.java
package believe.gui.testing;

import org.lwjgl.input.Cursor;
import org.newdawn.slick.Color;
import org.newdawn.slick.Font;
import org.newdawn.slick.Input;
import org.newdawn.slick.SlickException;
import org.newdawn.slick.gui.GUIContext;
import org.newdawn.slick.opengl.ImageData;

/**
 * Fake implementation of {@link GUIContext} for use within tests.
 */
public final class FakeGuiContext implements GUIContext {
  private static final Input INPUT = new Input(0);
  private static final Font DEFAULT_FONT = new Font() {
    @Override
    public int getWidth(String s) {
      return 0;
    }

    @Override
    public int getHeight(String s) {
      return 0;
    }

    @Override
    public int getLineHeight() {
      return 0;
    }

    @Override
    public void drawString(float v, float v1, String s) {

    }

    @Override
    public void drawString(float v, float v1, String s, Color color) {

    }

    @Override
    public void drawString(float v, float v1, String s, Color color, int i, int i1) {

    }
  };

  private final int width;
  private final int height;

  public FakeGuiContext(int width, int height) {
    this.width = width;
    this.height = height;
  }

  @Override
  public Input getInput() {
    return INPUT;
  }

  @Override
  public long getTime() {
    return 0;
  }

  @Override
  public int getScreenWidth() {
    return 0;
  }

  @Override
  public int getScreenHeight() {
    return 0;
  }

  @Override
  public int getWidth() {
    return width;
  }

  @Override
  public int getHeight() {
    return height;
  }

  @Override
  public Font getDefaultFont() {
    return DEFAULT_FONT;
  }

  @Override
  public void setMouseCursor(String s, int i, int i1) throws SlickException {

  }

  @Override
  public void setMouseCursor(ImageData imageData, int i, int i1) throws SlickException {

  }

  @Override
  public void setMouseCursor(Cursor cursor, int i, int i1) throws SlickException {

  }

  @Override
  public void setDefaultMouseCursor() {

  }
}

File: java/believe/gui/testing/FakeGuiContext.kt
package believe.gui.testing

import dagger.Reusable
import org.lwjgl.input.Cursor
import org.newdawn.slick.Color
import org.newdawn.slick.Font
import org.newdawn.slick.Input
import org.newdawn.slick.SlickException
import org.newdawn.slick.gui.GUIContext
import org.newdawn.slick.opengl.ImageData
import javax.inject.Inject

/** Fake implementation of [GUIContext] for use within tests.  */
@Reusable
class FakeGuiContext constructor(private val width: Int, private val height: Int) : GUIContext {
    @Inject
    internal constructor() : this(width = 0, height = 0)

    override fun getInput(): Input = INPUT

    override fun getTime() = 0L

    override fun getScreenWidth() = 0

    override fun getScreenHeight() = 0

    override fun getWidth(): Int = width

    override fun getHeight(): Int = height

    override fun getDefaultFont() = DEFAULT_FONT

    @Throws(SlickException::class)
    override fun setMouseCursor(s: String, i: Int, i1: Int) {
    }

    @Throws(SlickException::class)
    override fun setMouseCursor(
        imageData: ImageData, i: Int, i1: Int
    ) {
    }

    @Throws(SlickException::class)
    override fun setMouseCursor(
        cursor: Cursor, i: Int, i1: Int
    ) {
    }

    override fun setDefaultMouseCursor() {}

    companion object {
        private val INPUT = Input(0)
        private val DEFAULT_FONT: Font = object : Font {
            override fun getWidth(s: String) = 0

            override fun getHeight(s: String) = 0

            override fun getLineHeight() = 0

            override fun drawString(
                v: Float, v1: Float, s: String
            ) {
            }

            override fun drawString(
                v: Float, v1: Float, s: String, color: Color
            ) {
            }

            override fun drawString(
                v: Float, v1: Float, s: String, color: Color, i: Int, i1: Int
            ) {
            }
        }
    }

}